cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -960,7 +962,11 @@ writePoolState serialisedCurrentEpochState = do
           )
         )
 
-  liftIO . LBS.putStrLn $ encodePretty poolStates
+  let renderedResult = encodePretty poolStates
+  case mOutFile of
+    Nothing -> liftIO $ LBS.putStrLn renderedResult
+    Just (File oFp) -> handleIOExceptT (QueryCmdWriteFileError . FileIOError oFp)
+        $ LBS.writeFile oFp renderedResult
```

Comment:
Could you use `writeLazyByteStringOutput` to avoid the need for the `case`? You'll find example calls to this function elsewhere in this codebase :+1:

Reply:
Much nicer! Thank you.